### PR TITLE
Fixed a bug found by Chris Schneider

### PR DESCRIPTION
**Pull request by micheles**
https://ci.openquake.org/job/zdevel_oq-engine/1028

### CODE REVIEW

File: openquake/engine/calculators/risk/event_based_risk/core.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def event_based(workflow, getter, outputdict, params, monitor):
             if specific_assets:
                 loss_matrix, assets = _filter_loss_matrix_assets(
                     out.output.loss_matrix, out.output.assets, specific_assets)
-                if assets:
+                if len(assets):
```
Comment (daniviga) [inline openquake/engine/calculators/risk/event_based_risk/core.py]: This means that `assets` cannot be `None`?

Comment (larsbutler) [inline openquake/engine/calculators/risk/event_based_risk/core.py]: Yep; if `assets` is None, `len(assets)` will raise a `TypeError`.

**Comment by daniviga** [comment]
LGTM

**Comment by micheles** [comment]
The problem was that assets is a numpy array, so the syntax `if assets` gives an error. In the companion PR I have changed the risk demo so that the bug is exercised.

**Pull request merged**